src/core/ext/xds/xds_client_grpc.cc

```diff
@@ -178,6 +184,29 @@ absl::StatusOr<RefCountedPtr<GrpcXdsClient>> GrpcXdsClient::GetOrCreate(
   return xds_client;
 }
 
+grpc_slice GrpcXdsClient::DumpAllClientConfigs()
+    ABSL_NO_THREAD_SAFETY_ANALYSIS {
+  auto xds_client =
+      GrpcXdsClient::GetOrCreate(ChannelArgs(), "grpc_dump_xds_configs()");
+  upb::Arena arena;
+  // Following two containers should survive till serialization
+  std::set<std::string> string_pool;
+  auto response = envoy_service_status_v3_ClientStatusResponse_new(arena.ptr());
+  // If we aren't using xDS, just return a response with empty config.
```

Comment:
I don't think we actually want to add a `ClientConfig` to the response if there is no `XdsClient`.  I think we can just return an empty response proto in that case.  This is consistent with what we'll be doing once we do have multiple `XdsClient` instances.

Even if this is a change in the current behavior, I don't think it should cause any problems.